xdsl/backend/csl/print_csl.py

```diff
@@ -333,6 +382,24 @@ def print_block(self, body: Block):
                 case memref.GetGlobal(name_=name, memref=res):
                     # We print the array definition when the global is defined
                     self.variables[res] = name.string_value()
+                case memref.Store(value=val, memref=arr, indices=idxs):
+                    arr_name = self._get_variable_name_for(arr)
+                    idx_args = ", ".join(map(self._get_variable_name_for, idxs))
+                    val_name = self._get_variable_name_for(val)
+                    self.print(f"{arr_name}[{idx_args}] = {val_name};")
+                case memref.Load(memref=arr, indices=idxs, res=res):
+                    arr_name = self._get_variable_name_for(arr)
+                    idx_args = ", ".join(map(self._get_variable_name_for, idxs))
+                    # Use the array access syntax instead of copying the value out
+                    self.variables[res] = f"({arr_name}[{idx_args}])"
+                case csl.AddressOfOp(value=val, res=res):
+                    val_name = self._get_variable_name_for(val)
+                    ty = res.type
+                    assert isinstance(
+                        ty, csl.PtrType
+                    ), f"Result of {csl.AddressOfOp.name} has to be a pointer"
```

Comment:
This is checked by the `verify` (although I guess this is needed to make pyright happy?). Not sure we need the assert here, I think we could replace it by a `cast`, as it is never going to be wrong?

I guess this is a trade-off between more readable printing code vs. better error messages when printing invalid csl code...

I guess imo we don't need this assertion here.

Reply:
Yes, this assert is just for the sake of the type checker, it does get caught by `verify`. A `cast` should be safe here.